core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -95,6 +98,14 @@ protected void handleCommitConflict(Snapshot snapshot, Exception cause)
               CoreError.CONSENSUS_COMMIT_CONFLICT_OCCURRED_WHEN_COMMITTING_STATE.buildMessage(),
               cause,
               snapshot.getId());
+        } else if (state.equals(TransactionState.COMMITTED)) {
+          if (throwExceptionIfCommittedTransactionExists) {
+            throw new CommitConflictException(
```

Comment:
Let me ask about a corner case. If a proper transaction that started with the transaction ID and detected here the coordinator state conflict crashes for some reasons before rolling back PREPARE records, future lazy recoveries will roll-forward based on the unexpected (malicious?) commit?

Reply:
@komamitsu Umm, that’s a really good point… Regardless of crashing, the lazy recovery will roll forward the records based on the committed state of the unrelated transaction. That's the same thing as not throwing the exception after all... Let me think about it more.